compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -889,7 +889,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             self.dcx(),
                             errors::ExpectedReturnTypeLabel::Other { span: hir_ty.span, expected },
                         );
-                        self.try_suggest_return_impl_trait(err, expected, ty, fn_id);
+                        self.try_suggest_return_impl_trait(err, expected, found, fn_id);
```

Comment:
Actually I wonder if we should add this not in more cases, not just in cases where we suggest RPITs. For example here:

```rs
fn f<T>() -> (T,) {
    (0,)
}
```

Reply:
wait how do u undo a `r=` lol

Reply:
Okay, let's do this!